workflows/batch_workflow.py

```diff
@@ -0,0 +1,414 @@
+    'overwrite',
+    is_flag=True,
+    help='if an intermediate or a final file exists, skip running the code '
+    'that generates it.',
+)
+@click.option('--dry-run', 'dry_run', is_flag=True)
+@click.option('--run-vqsr/--skip-vqsr', 'run_vqsr', is_flag=True, default=True)
+@click.option('--run-rf/--skip-rf', 'run_rf', is_flag=True, default=True)
+def main(  # pylint: disable=too-many-arguments,too-many-locals,too-many-statements
+    callset_name: str,
+    callset_version: str,
+    callset_batches: List[str],
+    access_level: str,
+    input_bucket_suffix: str,
+    mt_output_bucket_suffix: str,
+    analysis_output_bucket_suffix: str,
+    skip_input_meta: bool,
+    keep_scratch: bool,
+    reuse_scratch_run_id: str,
+    dry_run: bool,
+    overwrite: bool,
+    run_vqsr: bool,
+    run_rf: bool,
+):
+    """
+    Drive a Hail Batch workflow that creates and submits jobs. A job usually runs
+    either a Hail Query script from the scripts folder in this repo using a Dataproc
+    cluster; or a GATK command using the GATK or Gnarly image.
+    """
+    logger.info(f'Enable random forest: {run_rf}')
+    logger.info(f'Enable VQSR: {run_vqsr}')
+
+    billing_project = os.getenv('HAIL_BILLING_PROJECT') or callset_name
+
+    if not mt_output_bucket_suffix:
+        if access_level == 'full':
+            mt_output_bucket_suffix = 'main'
+        else:
+            mt_output_bucket_suffix = 'temporary'
+
+    if not analysis_output_bucket_suffix:
+        if access_level == 'full':
+            analysis_output_bucket_suffix = 'analysis'
+        else:
+            analysis_output_bucket_suffix = 'temporary'
+
+    if not input_bucket_suffix:
+        if access_level in ['standard', 'full']:
+            input_bucket_suffix = 'main'
+        else:
+            input_bucket_suffix = 'test'
+
+    if not callset_batches:
+        if access_level == 'test':
+            callset_batches = ['0']
+        else:
+            raise click.BadParameter(
+                'Please, specify batch numbers with --batch '
+                '(can put multiple times, e.g. --batch 0 --batch 1)'
+            )
+
+    mt_output_bucket = f'gs://cpg-{callset_name}-{mt_output_bucket_suffix}/joint_vcf'
+    raw_combined_mt_path = join(mt_output_bucket, 'raw', 'genomes.mt')
+    # pylint: disable=unused-variable
+    filtered_combined_mt_path = join(mt_output_bucket, 'genomes.mt')
+
+    base_bucket = f'gs://cpg-{callset_name}-{analysis_output_bucket_suffix}/joint_vcf'
+    analysis_bucket = join(base_bucket, callset_version)
+    hail_bucket = os.environ.get('HAIL_BUCKET') or join(analysis_bucket, 'hail')
+    combiner_bucket = join(analysis_bucket, 'combiner')
+
+    backend = hb.ServiceBackend(
+        billing_project=billing_project,
+        bucket=hail_bucket.replace('gs://', ''),
+    )
+    b = hb.Batch('Joint Calling', backend=backend)
+    scripts_dir = abspath(join(dirname(__file__), '..', 'scripts'))
+
+    # TODO: merge with existing data
```

Comment:
Might be worth tracking that as an issue, but I guess it's really also a part of the upload processor.

Reply:
Working with Vivian on that!